### PR TITLE
feat(pipes): improved kebab-case pipe

### PR DESCRIPTION
**Pull request by Xinchro**
# Description

Updated the KebabCase pipe to include a wider variety of cases to kebab.

## Requirements

Automation should cover the testing, outside of disagreeing with test cases.

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [x] I have provided an adequate amount of test coverage
- [ ] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [ ] I have provided a story in storybook to document the changes
- [ ] I have added the documentation
- [ ] I have added any new public feature modules to public-api.ts

### CODE REVIEW

File: projects/canopy/src/lib/pipes/kebab-case/kebab-case.pipe.ts [hunk 1]
```diff
@@ -7,6 +7,13 @@ export class LgKebabCasePipe implements PipeTransform {
   transform(str: string): any {
     return !str
       ? str
-      : str.toLowerCase().replace(/\s/g, '-');
+      : str
+        .replace(/[_\-|]{1,}/g, ' ')
+        .replace(/([a-z])([A-Z])/g, function(match, group1, group2) {
```
Comment (elenagarrone) [inline projects/canopy/src/lib/pipes/kebab-case/kebab-case.pipe.ts:12]: Could you write it as an arrow function please? `.replace(/([a-z])([A-Z])/g, (match, group1, group2) => {`

File: projects/canopy/src/lib/pipes/kebab-case/kebab-case.pipe.ts [hunk 2]
```diff
@@ -7,6 +7,13 @@ export class LgKebabCasePipe implements PipeTransform {
...
+          return `${group1} ${group2.toLowerCase()}`;
```
Comment (elenagarrone) [inline projects/canopy/src/lib/pipes/kebab-case/kebab-case.pipe.ts:13]: Shouldn't the whole string be made lowercase?

Comment (Xinchro) [reply projects/canopy/src/lib/pipes/kebab-case/kebab-case.pipe.ts:13]: `group1` and `group2` here are just 2 characters(a lower case and an upper case, respectively), the whole thing is also lowercased at the end (`toLowerCase()`)

File: projects/canopy/src/lib/pipes/kebab-case/kebab-case.pipe.ts (hunk 1 above)
Comment (Xinchro) [reply projects/canopy/src/lib/pipes/kebab-case/kebab-case.pipe.ts]: yep, done

**Comment by elenagarrone** [comment]
Looks good on storybook.

**Review by elenagarrone — APPROVED** [review]

**Review by NemirStephan — APPROVED** [review]

**Comment by elenagarrone** [comment]
FEP-1648

**Comment by elenagarrone** [comment]
FEP-1648

**Pull request merged**